devtools/create_project/msbuild.cpp

```diff
@@ -80,61 +81,67 @@ void MSBuildProvider::createProjectFile(const std::string &name, const std::stri
 	           "<Project DefaultTargets=\"Build\" ToolsVersion=\"" << _msvcVersion.project << "\" xmlns=\"http://schemas.microsoft.com/developer/msbuild/2003\">\n"
 	           "\t<ItemGroup Label=\"ProjectConfigurations\">\n";
 
-	outputConfiguration(project, "Debug", "Win32");
-	outputConfiguration(project, "Debug", "x64");
-	outputConfiguration(project, "Analysis", "Win32");
-	outputConfiguration(project, "Analysis", "x64");
-	outputConfiguration(project, "LLVM", "Win32");
-	outputConfiguration(project, "LLVM", "x64");
-	outputConfiguration(project, "Release", "Win32");
-	outputConfiguration(project, "Release", "x64");
+	std::array<MSVC_Architecture, 3> archs{ MSVC_Architecture::ARCH_X86, MSVC_Architecture::ARCH_AMD64, MSVC_Architecture::ARCH_ARM64 };
```

Comment:
`std::list` makes more sense in my opinion. I am not sure c++11 initializers will work on all platforms.

Reply:
Why list? List is usually implemented work a doubly-linked list, you'd want to use it for low accumulated cost of insertion/deletion, but iteration takes O(n) time. Array on the other hand encapsulates fixed size arrays, which is exactly the use case here. I know there are only three elements here, but why take worse path?

Reply:
Your arrays are less than 10 elements, and you always move forward and never change the elements. By using list you will avoid adding another #include in the code which will lead to lesser code footprint.

Reply:
I disagree with your argument completely. There is still cost O(n) even for 3 elements, but if you want bad code, I implemented it as you requested.

> By using list you will avoid adding another #include in the code which will lead to lesser code footprint.

I would more willing to heed this advice if there weren't so many include files with full implementations in them.